integreat_cms/api/v3/chat/zammad_api.py

```diff
@@ -149,7 +149,13 @@ def get_messages(self, chat: UserChat) -> dict[str, dict | list[dict]]:
         :param chat: UserChat instance for the relevant Zammad ticket
         """
         response = self._parse_response(
-            self._attempt_call(self.client.ticket.articles, chat.zammad_id)
+            [
+                article
+                for article in self._attempt_call(
+                    self.client.ticket.articles, chat.zammad_id
+                )
+                if not article.get("internal")
+            ]
```

Comment:
`_attempt_call` can either return a list of response items, or a single error dict. In the latter case, that dict gets flattened into a list of its keys by this list comprehension 😅 

Actually, the code I wrote right below this is also guilty of this. So even though I dislike the `isinstance()` pattern, I think we need to change the entire function to (untested!!):
```python
# pylint: disable=method-hidden
    def get_messages(self, chat: UserChat) -> dict[str, dict | list[dict]]:
        """
        Get all messages for a given ticket

        :param chat: UserChat instance for the relevant Zammad ticket
        """
        raw_response = self._attempt_call(self.client.ticket.articles, chat.zammad_id)
        if not isinstance(raw_response, list):
            return self._parse_response(raw_response)  # type: ignore[return-value]

        response = self._parse_response(
            [article for article in raw_response if not article.get("internal")]
        )
        for message in response:
            if "attachments" in message:
                message["attachments"] = [
                    self._transform_attachment(chat, message["id"], attachment)
                    for attachment in message["attachments"]
                ]

        return {"messages": response}
```